dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -602,4 +604,44 @@ public static Fence newLwSyncBarrier() {
         }
     }
 
-}
+    // =============================================================================================
+    // ============================================ PTX ============================================
+    // =============================================================================================
+    public static class PTX {
+        private PTX() {}
+
+        public static Store newTaggedStore(IExpr address, ExprInterface value, String mo,  String scope) {
+            Store store = new Store(address, value, mo); // mo = REL || RLX || WEAK
+            store.addFilters(scope); // scope = CTA || GPU || SYS
+            return store;
+        }
+
+        public static Load newTaggedLoad(Register register, IExpr address, String mo, String scope) {
+            Load load = new Load(register, address, mo); // mo = ACQ || RLX || WEAK
+            load.addFilters(scope); // scope =  CTA || GPU || SYS
+            return load;
+        }
+
+        public static Fence newTaggedFence(String mo, String scope) {
```

Comment:
This method seems unnecessary. You can simply create a normal fence and after creating the event, add the scope tag.

The same is true for the store and load above

Reply:
Accepted